### PR TITLE
Show image history

### PR DESCRIPTION
**Pull request by jelly**
A new History tab has been added to the images view which fetches the
history on demand.

The comment field always seems to be empty.. and therefore cut off.

@garrett please take a look:

image

image

On Mobile

image

On tablet

image

### CODE REVIEW

**Comment by garrett** [comment]
> The comment field always seems to be empty.. and therefore cut off.

Are we able to see if there are any comments and hide the column if there aren't any?

And can we not show anything if it's just `<missing>` in the ID?

**Comment by jelly** [comment]
Found one with comments:


image

**Comment by garrett** [comment]
> Padding comes from PF, it's not really double padding:

I understand that... but it is literally double padding. There's padding, a frame, and padding again. I'm not talking about individual HTML elements, but from a layout / design standpoint. There shouldn't be. We already have padding and a frame, we don't need two borders with padding with each doubled up for the same element. It's just way too much in mobile mode and the data is constrained by all the redundant space and borders.

It's because we're nesting multiple elements within each other.

This is one of the (many) reasons we made the Machines page has sub-pages, not expandable tables with tabs and stuff like in Cockpit-Podman.

(But we had and still have some CSS hacks to strip away the redundant padding and borders around Cockpit.)

**Comment by jelly** [comment]
What do you think of:

image

File: src/ImageHistory.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import * as client from './client.js';
+
+import { ListingTable } from "cockpit-components-table.jsx";
+
+const _ = cockpit.gettext;
+
+const IdColumn = Id => {
+    Id = utils.truncate_id(Id);
+    // Not an id but <missing> or something else
+    if (Id.length != 12) {
```
Comment (marusak) [inline src/ImageHistory.jsx:13]: Not the most robust check, but I don't have any better idea either (except maybe additionally checking if it is also hexadecimal)

File: src/ImageHistory.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    return Id;
+};
+
+const ImageDetails = ({ image }) => {
+    const [history, setHistory] = useState([]);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        client.imageHistory(image.isSystem, image.Id).then(history => setHistory(history))
```
Comment (marusak) [inline src/ImageHistory.jsx:24]: you can just pass `setHistory`, no need for the function wrapper.
`client.imageHistory(image.isSystem, image.Id).then(setHistory)`

**Review by marusak — CHANGES_REQUESTED** [review]
Nice, thanks!

Tests are unhappy, two small comments and please add release note, thanks!

File: src/ImageHistory.jsx (hunk 1 above)
Comment (jelly) [reply src/ImageHistory.jsx:13]: An idea is to use regex, like `<[a-z]+>`

File: src/ImageHistory.jsx (hunk 2 above)
Comment (jelly) [reply src/ImageHistory.jsx]: Good point!

File: src/ImageHistory.jsx (hunk 1 above)
Comment (jelly) [reply src/ImageHistory.jsx]: Applied, seems to work nicely

**Comment by jelly** [comment]
> Nice, thanks!
> 
> Tests are unhappy, two small comments and please add release note, thanks!

Test should be fixed, React keeps a tbody around from the History tab.. which is kinda annoying. Also added a pixel test for the history table

**Review by marusak — APPROVED** [review]
Nice, thank you!

**Pull request merged**